lib/shopify-cli/commands/connect.rb

```diff
@@ -4,16 +4,30 @@ module ShopifyCli
   module Commands
     class Connect < ShopifyCli::Command
       def call(*)
-        if Project.current
-          @ctx.puts @ctx.message('core.connect.production_warning')
-          org = fetch_org
-          id = org['id']
-          app = get_app(org['apps'])
-          shop = get_shop(org['stores'], id)
-          write_env(app, shop)
-          @ctx.puts(@ctx.message('core.connect.connected', app.first['title']))
-          @ctx.puts(@ctx.message('core.connect.serve', ShopifyCli::TOOL_NAME))
+        project_type = ask_project_type unless Project.has_current?
+
+        @ctx.puts @ctx.message('core.connect.production_warning') if Project.has_current? && Project.current
+
+        env_file = Resources::EnvFile.parse_external_env
+        org = fetch_org
+        id = org['id']
+        app = get_app(org['apps'])
+        shop = get_shop(org['stores'], id)
+
+        write_env(app, shop, env_file[:scopes], env_file[:extra])
+        write_cli_yml(project_type, id) unless Project.has_current?
+
+        @ctx.puts(@ctx.message('core.connect.connected', app.first['title']))
+        @ctx.puts(@ctx.message('core.connect.serve', ShopifyCli::TOOL_NAME))
+      end
+
+      def ask_project_type
+        type_name = CLI::UI::Prompt.ask(@ctx.message("core.connect.project_type_select")) do |handler|
+          ShopifyCli::Commands::Create.all_visible_type.each do |type|
+            handler.option(type.project_name) { type.project_type }
+          end
         end
+        type_name
```

Comment:
Do we need this intermediate variable?

Reply:
haha no, I thought it made it more readable to show that we get something from the prompt, but I can get rid of `type_name` entirely and it still works